scripts/compare-fscontent/compare-fscontent.go

```diff
@@ -0,0 +1,158 @@
+	}
+	inv := invoker.New(c, nil)
+
+	nnsReader := nns.NewReader(inv, nnsState.Hash)
+	containerH, err := nnsReader.ResolveFSContract(nns.NameContainer)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to resolve container contract: %w", err)
+	}
+	reader := container.NewReader(inv, containerH)
+	containers, err := reader.List([]byte{})
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to list containers: %w", err)
+	}
+
+	netmapH, err := nnsReader.ResolveFSContract(nns.NameNetmap)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to resolve netmap contract: %w", err)
+	}
+	netmapReader := netmap.NewReader(inv, netmapH)
+	netmap, err := netmapReader.Netmap()
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to retrieve netmap: %w", err)
+	}
+	return containers, netmap, nil
+}
+
+func cliMain(c *cli.Context) error {
+	a := c.Args().Get(0)
+	b := c.Args().Get(1)
+	if a == "" {
+		return errors.New("no arguments given")
+	}
+	if b == "" {
+		return errors.New("missing second argument")
+	}
+	ca, ha, err := initClient(a, "A")
+	if err != nil {
+		return err
+	}
+	cb, hb, err := initClient(b, "B")
+	if err != nil {
+		return err
+	}
+	if ha != hb {
+		var diff = hb - ha
+		if ha > hb {
+			diff = ha - hb
+		}
+		if diff > 10 && !c.Bool("ignore-height") { // Allow some height drift.
+			return fmt.Errorf("chains have different heights: %d vs %d", ha, hb)
+		}
+	}
+	fmt.Printf("RPC %s hight: %d\nRPC %s height: %d\n", a, ha, b, hb)
+
+	containersA, netmapA, err := getFSContent(ca)
+	if err != nil {
+		return fmt.Errorf("RPC %s: %w", a, err)
+	}
+	containersB, netmapB, err := getFSContent(cb)
+	if err != nil {
+		return fmt.Errorf("RPC %s: %w", b, err)
+	}
+
+	if len(containersA) != len(containersB) {
+		return fmt.Errorf("number of containers mismatch: %d vs %d", len(containersA), len(containersB))
+	}
+	fmt.Printf("number of containers checked: %d\n", len(containersA))
+	for i := range containersA {
+		if !bytes.Equal(containersA[i], containersB[i]) {
+			dumpContentDiff("container", i, a, b, containersA[i], containersB[i])
+		}
+	}
+
+	if len(netmapA) != len(netmapB) {
+		return fmt.Errorf("number of netmap entries mismatch: %d vs %d", len(netmapA), len(netmapB))
+	}
+	fmt.Printf("number of netmap entries checked: %d\n", len(netmapA))
+	for i := range netmapA {
+		if netmapA[i].State.Cmp(netmapB[i].State) != 0 || !bytes.Equal(netmapA[i].BLOB, netmapB[i].BLOB) {
+			dumpContentDiff("netmap entry", i, a, b, netmapA[i], netmapB[i])
+		}
+	}
+	return nil
+}
+
+func dumpContentDiff(itemName string, i int, a string, b string, itemA any, itemB any) {
+	fmt.Printf("%s %d:\n", itemName, i)
+	da := spew.Sdump(itemA)
+	db := spew.Sdump(itemB)
+	diff, _ := difflib.GetUnifiedDiffString(difflib.UnifiedDiff{
+		A:        difflib.SplitLines(da),
+		B:        difflib.SplitLines(db),
+		FromFile: a,
+		FromDate: "",
+		ToFile:   b,
+		ToDate:   "",
+		Context:  1,
+	})
+	fmt.Println(diff)
+}
+
+func main() {
+	ctl := cli.NewApp()
```

Comment:
"neofs" things usually use https://github.com/spf13/cobra. i do not mind the other things in general but why is it placed here?

Reply:
> i do not mind the other things in general

This script is ported from NeoGo as is, so if you don't mind, then let's keep it as is.

> why is it placed here?

Any other suggestions? It's not NeoGo related, and it brings additional cyclic dependency from neofs-contract to NeoGo. And it's also not a part of NeoFS. It's directly related to the contracts since it's intended to check the contracts' storage.